windows/src/desktop/kmshell/main/OnlineUpdateCheck.pas

```diff
@@ -339,6 +463,128 @@ function TOnlineUpdateCheck.DownloadUpdates: Boolean;
+        Proxy.Password := GetProxySettings.Password;
+        Request.Agent := API_UserAgent;
+
+        Request.SetURL(url);
+        Upload;
+        if Response.StatusCode = 200 then
+        begin
+          with TFileStream.Create(savepath, fmCreate) do
+          try
+            Write(Response.PMessageBody^, Response.MessageBodyLength);
+          finally
+            Free;
+          end;
+          Result := True;
+        end
+        else // I2742
+          // If it fails we set to false but will try the other files
+          Result := False;
+          Exit;
+      finally
+        Free;
+      end;
+    end;
+
+
+begin
+  Result := False;
+  try
+    FDownload.TotalSize := 0;
+    FDownload.TotalDownloads := 0;
+    downloadCount := 0;
+
+    for i := 0 to High(FParams.Packages) do
+      if FParams.Packages[i].Install then
+      begin
+        Inc(FDownload.TotalDownloads);
+        Inc(FDownload.TotalSize, FParams.Packages[i].DownloadSize);
+
+        FParams.Packages[i].SavePath := SavePath + FParams.Packages[i].FileName;
+      end;
+
+    if FParams.Keyman.Install then
+    begin
+      Inc(FDownload.TotalDownloads);
+      Inc(FDownload.TotalSize, FParams.Keyman.DownloadSize);
+      FParams.Keyman.SavePath := SavePath + FParams.Keyman.FileName;
+    end;
+
+    FDownload.StartPosition := 0;
+    for i := 0 to High(FParams.Packages) do
+      if FParams.Packages[i].Install then
+      begin
+        if not DownloadFile(FParams.Packages[i].DownloadURL, FParams.Packages[i].SavePath) then // I2742
+        begin
+          FParams.Packages[i].Install := False; // Download failed but install other files
+        end
+        else
+          Inc(downloadCount);
+        FDownload.StartPosition := FDownload.StartPosition + FParams.Packages[i].DownloadSize;
+      end;
+
+    if FParams.Keyman.Install then
+      if not DownloadFile(FParams.Keyman.DownloadURL, FParams.Keyman.SavePath) then  // I2742
+      begin
+        FParams.Keyman.Install := False;  // Download failed but user wants to install other files
+      end;
+
+    // There needs to be at least one file successfully downloaded to return
+    // TRUE that files where downloaded
+    if downloadCount > 0  then
+      Result := True;
+  except
+    on E:EHTTPUploader do
+    begin
+      if (E.ErrorCode = 12007) or (E.ErrorCode = 12029)
+        then KL.Log(S_OnlineUpdate_UnableToContact)
+        else KL.Log(WideFormat(S_OnlineUpdate_UnableToContact_Error, [E.Message]));
+      Result := False;
+    end;
+  end;
+end;
+
+function TOnlineUpdateCheck.DownloadUpdatesBackground: Boolean;
+var
+  i: Integer;
+  DownloadBackGroundSavePath : String;
+  DownloadResult : Boolean;
+begin
+  //DownloadTempPath := IncludeTrailingPathDelimiter(CreateTempPath);
+  DownloadBackGroundSavePath := IncludeTrailingPathDelimiter(GetFolderPath(CSIDL_COMMON_APPDATA) + SFolder_CachedUpdateFiles);
+  //DownloadBackGroundSavePath :=  DownloadBackGroundSavePath + 'RCTest.txt';
```

Comment:
The commented lines look like they are left over from development and thus should be removed?